### ISSUE TITLE
segfault when using vo_dmabuf_wayland

### ISSUE THREAD

**Issue by llyyr** [mpv-player/mpv#11442]
### Important Information

Provide following Information:
- mpv version: Tested with git head, also with the initial vo_dmabuf_wayland commit, and some random commits in between
- Linux Distribution and Version: OpenSUSE Tumbleweed
- Window Manager and version: Tested with sway 1.8.1 + wlroots 0.16.2, as well as sway git and wlroots git
- GPU driver and version: 6600 XT + Mesa 23.0.0

### Reproduction steps

mpv --no-config --vo=dmabuf-wayland [video]

This used to work on my older PC with only a really old Intel iGPU, but it doesn't on this PC and it's my first time trying to use it so I can't say if it's a regression from some update...

### Expected behavior

Video plays

### Actual behavior

mpv segfaults with errors in logs about hwupload surface pixel format.

### Log file

[mpv2.log](https://github.com/mpv-player/mpv/files/10950026/mpv2.log)

**Comment by llyyr** [comment]
Stack trace: [trace.log](https://github.com/mpv-player/mpv/files/10950042/trace.log)

**Comment by Dudemanguy** [comment]
>AVHWFramesContext: Created surface 0x1.

Hmm, that looks quite suspicious. Does anything change if you specify hwdec (i.e. `--hwdec=vaapi`)?

**Comment by llyyr** [comment]
Hmm, I had initially overlooked something when testing. If `WLR_RENDERER=vulkan` is set, mpv does not segfault. It just flickers and drops a ton of frames, and goes out of sync with the audio.

With the gles2 renderer, mpv always segfaults when specifying `hwdec=vaapi`, and displays a single grayscale frame before segfaulting without specifying vaapi.

I don't know if all this info is relevant or not, but mpv is built with

```
libavutil      57. 28.100 / 57. 28.100
libavcodec     59. 37.100 / 59. 37.100
libavformat    59. 27.100 / 59. 27.100
libavdevice    59.  7.100 / 59.  7.100
libavfilter     8. 44.100 /  8. 44.100
libswscale      6.  7.100 /  6.  7.100
libswresample   4.  7.100 /  4.  7.100
libpostproc    56.  6.100 / 56.  6.100
```

And I have a dual gpu system with 6600 XT and the processor's iGPU.

**Comment by Dudemanguy** [comment]
>although mpv probably still shouldn't segfault?

Indeed, although it's hard to tell at a glance why exactly this occurs. Probably we're just trying to call `av_hwframe_transfer_data` on something that we shouldn't.

**Comment by llyyr** [comment]
The segfault was already fixed by some other commit before, but it used to result in flickering and video corruptions. Properly fixed by #12048
